Assets/Standard Assets/Characters/FirstPersonCharacter/Scripts/FirstPersonController.cs

```diff
@@ -117,6 +140,15 @@ private void FixedUpdate()
             m_MoveDir.x = desiredMove.x*speed;
             m_MoveDir.z = desiredMove.z*speed;
 
+            if (isPlayerKeyLocXPositive && m_MoveDir.x < 0 || !isPlayerKeyLocXPositive && m_MoveDir.x > 0)
+            {
+                m_MoveDir.x *= keyLocationSpeedMod;
+            }
+            if (isPlayerKeyLocZPositive && m_MoveDir.z < 0 || !isPlayerKeyLocZPositive && m_MoveDir.z > 0)
+            {
+                m_MoveDir.z *= keyLocationSpeedMod;
+            }
+
```

Comment:
```suggestion
            if (IsSomethingHappening1)
            {
                m_MoveDir.x *= keyLocationSpeedMod;
            }
            if (IsSomethingHappening2)
            {
                m_MoveDir.z *= keyLocationSpeedMod;
            }
```
Can't understand a thing. Since the values that you're checking in those `if` sentences are very arbitrary (I mean e.g. `isPlayerKeyLocZPositive` says kind of nothing to me), these could be put to functions, which would explain better what's happening here.